fms_fsdp/config/training.py

```diff
@@ -20,7 +20,7 @@ class train_config:
     # fsdp policies
     mixed_precision: bool = True
     fsdp_activation_checkpointing: bool = False
-    selective_checkpointing: int = 1
+    selective_checkpointing: float | str = 1  # percentage of blocks to apply ac
```

Comment:
what's the purpose of allowing a string here?

Reply:
@nairbv to support fraction format.

With a single value percentage p, a common practice/preference would be using a fraction, comparing to floating number.  e.g. I want to do p=1/3 (instead of writing p=0.33333) or p=3/7 (instead of p=0.428571428). When passing fraction (in command line for fire.Fire, or simply torchrun), it is interpreted as string.

During runtime this will be casted to float with `eval()` and `float()`.

Reply:
ah I see. on the `eval` line a comment like "for fractions in `Fire` command line parameters" might make it the intent clearer

Reply:
@nairbv good point. Added.